### PR TITLE
docs: use `@vite-pwa/vitepress`

### PR DESCRIPTION
**Pull request by userquin**
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

This PR replaces `vite-plugin-pwa` with `@vite-pwa/vitepress` package and also includes:
- remove post docs script, rn the og images generated using [transformHead]() VP build hook (removes the og images from the app configuration and adds the pair generating the corresponding og image inside the VP build process)
- sw moved from `prompt` to `autoUpdate` strategy and using`registerType: 'inline'`: we don't have auto page reload once the sw is activated, we'll need to include `workbox-window`, ping me if you want to add it
- sw registration now done by `@vite-pwa/vitepress`  integration
- include svg, woff2, ico, and some png to the sw precache manifest: excluded all social media images `og-*.png` generated in the `transformHead`
- changed pwa and favicon icons and included maskable and apple-touch-icon: generated on my local using `@vite-pwa/assets-generator` with Vueuse's `favicon.svg`: changed also links for new favicon and included apple-touch-icon ~~(I need to cleanup VP config file, awaiting @antfu confirmation about including `workbox-window`)~~
- included experimental `includeAllowlist` to avoid breaking the layout when requesting missing page: moved logic to new custom sw (I need to remove `workbox` entry)
- included 4 new runtime caching handler for google fonts and css, github avatars and sponsors (using network first with 7 days of expiration)
- use custom sw to include auto page reload when new sw installed and activated: I need to remove `workbox` entry from pwa configuration (testing auto page reload on PR preview, on my local working preperly)

### CODE REVIEW

**Comment by userquin** [comment]
there is no PR preview?

**Comment by antfu** [comment]
I disabled the deploy preview before and just brought it back. You cam try to push a random commit and trigger it.

File: packages/.vitepress/config.ts [hunk 1]
```diff
@@ -167,25 +170,145 @@ export default defineConfig({
...
+            },
+          },
+        },
+        {
+          urlPattern: /^https:\/\/avatars\.githubusercontent\.com\/.*/i,
+          handler: 'NetworkFirst',
+          options: {
+            cacheName: 'avatars-images-cache',
+            expiration: {
+              maxEntries: 10,
```
Comment (userquin) [inline packages/.vitepress/config.ts:284]: Increase this number

Comment (userquin) [reply packages/.vitepress/config.ts:291]: I'm going to remove this handler, we've a lot of contributors and the avatars from gh, we're wasting space due to crossorigin and opaque padding

**Review by antfu — APPROVED** [review]

**Pull request merged**